api/client/beacon/client.go

```diff
@@ -93,60 +95,47 @@ func WithTimeout(timeout time.Duration) ClientOpt {
 
 // Client provides a collection of helper methods for calling the Eth Beacon Node API endpoints.
 type Client struct {
-	hc     *http.Client
-	host   string
-	scheme string
+	hc      *http.Client
+	host    string
+	scheme  string
+	baseURL *url.URL
 }
 
 // NewClient constructs a new client with the provided options (ex WithTimeout).
 // `host` is the base host + port used to construct request urls. This value can be
 // a URL string, or NewClient will assume an http endpoint if just `host:port` is used.
 func NewClient(host string, opts ...ClientOpt) (*Client, error) {
-	host, err := validHostname(host)
+	u, err := urlForHost(host)
 	if err != nil {
 		return nil, err
 	}
 	c := &Client{
-		hc:     &http.Client{},
-		scheme: "http",
-		host:   host,
+		hc:      &http.Client{},
+		baseURL: u,
 	}
 	for _, o := range opts {
 		o(c)
 	}
 	return c, nil
 }
 
-func validHostname(h string) (string, error) {
+func urlForHost(h string) (*url.URL, error) {
```

Comment:
This now returns a parsed `*url.URL` instead of a subset of URL attributes, to ensure there is no information loss from the user input when rendering paths for API calls.